osf-cli/src/main/java/org/dataconservancy/cos/packaging/cli/PackageGenerationApp.java

```diff
@@ -101,13 +142,96 @@ public static void main(String[] args) {
 			System.exit(1);
 		} catch (Exception e){
             System.err.println(e.getMessage());
-           // System.exit(e.getCode());
+            System.exit(1);
         }
     }
 
    	private void run() throws Exception {
-        //boolean useDefaults = true;
-        System.err.println("MOOOOOOOOOOOOOO");
+        final ClassPathXmlApplicationContext cxt =
+                new ClassPathXmlApplicationContext("classpath*:applicationContext.xml",
+                    "classpath*:org/dataconservancy/config/applicationContext.xml",
+                    "classpath*:org/dataconservancy/packaging/tool/ser/config/applicationContext.xml",
+                    "classpath*:org/dataconservancy/cos/osf/client/config/applicationContext.xml",
+                    "classpath:/org/dataconservancy/cos/packaging/config/applicationContext.xml");
+
+        final OsfPackageGraph packageGraph = cxt.getBean("packageGraph", OsfPackageGraph.class);
+        final OsfService osfService = cxt.getBean("osfService", OsfService.class);
+        IpmPackager ipmPackager = new IpmPackager();
+
+        ipmPackager.setPackageName(bagName);
+
+        final Registration registration = osfService.registrationByUrl(registrationUrl).execute().body();
+        final List<User> users = registration.getContributors().stream()
+                .map(c -> {
+                    try {
+                        if (c.getUserRel() != null) {
+                            return osfService.userByUrl(c.getUserRel()).execute().body();
+                        } else {
+                            String contributorId = c.getId();
+                            if (contributorId.contains("-")) {
+                                contributorId = contributorId.split("-")[1];
+                            }
+                            return osfService.user(contributorId).execute().body();
+                        }
+                    } catch (IOException e) {
+                        throw new RuntimeException(e.getMessage(), e);
+                    }
+                })
+                .collect(Collectors.toList());
+
+        packageGraph.add(registration);
+        users.forEach(packageGraph::add);
+
+        LinkedHashMap<String, List<String>> metadata = createPackageMetadata();
+
+        Package pkg = ipmPackager.buildPackage(packageGraph, metadata);
+
+        /* Now just write the package out to a file in the output location*/
+        File packageFile = new File(outputLocation.getAbsolutePath(), bagName + ".tar.gz");
```

Comment:
Is hard-coding the extension what we want to do here?  I'm not sure if we are supporting multiple output serializations.  If not, then hard-coding is fine with me.